imagepipeline/src/main/java/com/facebook/imagepipeline/nativecode/DalvikPurgeableDecoder.java

```diff
@@ -114,10 +123,14 @@ protected DalvikPurgeableDecoder() {
       Bitmap.Config bitmapConfig,
       @Nullable Rect regionToDecode,
       int length,
-      final boolean transformToSRGB) {
+      @Nullable final ColorSpace colorSpace) {
     BitmapFactory.Options options = getBitmapFactoryOptions(
         encodedImage.getSampleSize(),
         bitmapConfig);
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      options.inPreferredColorSpace = colorSpace == null
```

Comment:
This check (and the one above) are going to break DEX preverification. Can you instead move these to a util class, similar to this:

```
    ...
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
      OreoUtils.setColorSpace(options, colorSpace);
    }
    ...
  }



  ....
  import com.facebook.annotations.DoNotOptimize;
  @DoNotOptimize
  private static class OreoUtils {
    void setColorSpace(...) {
        options.inPreferredColorSpace = ...
    }
  }
}
```